assets/js/components/OverlayNotification/AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification.js

```diff
@@ -0,0 +1,247 @@
+	const ga4AdSenseIntegrationEnabled = useFeature( 'ga4AdSenseIntegration' );
+	const breakpoint = useBreakpoint();
+
+	const dashboardType = useDashboardType();
+	const isMainDashboard = dashboardType === DASHBOARD_TYPE_MAIN;
+
+	const isShowingNotification = useSelect( ( select ) =>
+		select( CORE_UI ).isShowingOverlayNotification(
+			ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const { setOverlayNotificationToShow, dismissOverlayNotification } =
+		useDispatch( CORE_UI );
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed(
+			ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const isDismissing = useSelect( ( select ) =>
+		select( CORE_USER ).isDismissingItem(
+			ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const analyticsModuleConnected = useSelect( ( select ) => {
+		if ( ! isMainDashboard || isDismissed ) {
+			return null;
+		}
+		return select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+	} );
+
+	const adSenseModuleConnected = useSelect( ( select ) => {
+		if ( ! isMainDashboard || isDismissed ) {
+			return null;
+		}
+		return select( CORE_MODULES ).isModuleConnected( 'adsense' );
+	} );
+
+	const canViewSharedAnalytics = useSelect( ( select ) => {
+		if ( ! isMainDashboard || isDismissed ) {
+			return null;
+		}
+		return select( CORE_USER ).hasAccessToShareableModule( 'analytics-4' );
+	} );
+	const canViewSharedAdSense = useSelect( ( select ) => {
+		if ( ! isMainDashboard || isDismissed ) {
+			return null;
+		}
+		return select( CORE_USER ).hasAccessToShareableModule( 'adsense' );
+	} );
+
+	const isAdSenseLinked = useSelect( ( select ) => {
+		if ( ! isMainDashboard || isDismissed ) {
+			return null;
+		}
+
+		return select( MODULES_ANALYTICS_4 ).getAdSenseLinked();
+	} );
+
+	const adSenseAccountID = useSelect( ( select ) => {
+		if ( adSenseModuleConnected ) {
+			return select( MODULES_ADSENSE ).getAccountID();
+		}
+
+		return null;
+	} );
+
+	const { startDate, endDate } = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const args = {
+		startDate,
+		endDate,
+		dimensions: [ 'pagePath', 'adSourceName' ],
+		metrics: [ { name: 'totalAdRevenue' } ],
+		filter: {
+			fieldName: 'adSourceName',
+			stringFilter: {
+				matchType: 'EXACT',
+				value: `Google AdSense account (${ adSenseAccountID })`,
+			},
+		},
+		orderby: [ { metric: { metricName: 'totalAdRevenue' }, desc: true } ],
+		limit: 1,
+	};
+
+	const data = useSelect( ( select ) => {
+		if (
+			ga4AdSenseIntegrationEnabled &&
+			isMainDashboard &&
+			isDismissed === false &&
+			isAdSenseLinked &&
+			adSenseModuleConnected &&
+			analyticsModuleConnected &&
+			canViewSharedAdSense &&
+			canViewSharedAnalytics
+		) {
+			return select( MODULES_ANALYTICS_4 ).getReport( args );
+		}
+
+		return null;
+	} );
+
+	const dataAvailable = !! data?.rows?.length;
```

Comment:
Not sure actually, I think I've followed the internal implementation from the PieChart component since I was following the trail from the Top Earning Pages widget. I've updated this to use the utility function and all seems well. :crossed_fingers: